tests/e2e/kcs/test_selinux_relabel_solution.py

```diff
@@ -0,0 +1,247 @@
+            timeout=300,
+        )
+        log.info(f"files are {random_file}")
+        ini_md5sum_pod_data = res_pod.cal_md5sum(pod_obj=pod_obj, file_name=random_file)
+        pod_objs = res_pod.get_all_pods(
+            namespace=namespace,
+            selector=[self.pod_selector],
+            selector_label="deploymentconfig",
+        )
+
+        # Respin pod
+        for pod in pod_objs:
+            pod1_name = self.pod_selector + "-1-deploy"
+            pod.ocp.wait_for_delete(resource_name=pod1_name, timeout=360)
+            pod.delete(wait=True)
+
+        assert wait_for_pods_to_be_running(timeout=600, sleep=15)
+        pod_objs = res_pod.get_all_pods(
+            namespace=namespace,
+            selector=[self.pod_selector],
+            selector_label="deploymentconfig",
+        )
+        for pod_obj in pod_objs:
+            pod_name = pod_obj.name
+            if "-1-deploy" not in pod_name:
+                pod_obj1 = pod_obj
+            fin_md5sum_pod_data = res_pod.cal_md5sum(
+                pod_obj=pod_obj1, file_name=random_file
+            )
+            assert ini_md5sum_pod_data == fin_md5sum_pod_data
+
+    def teardown(self):
+        """
+        Cleanup the test environment
+        """
+        res_pod.delete_deploymentconfig_pods(self.pod_objs[0])
+        self.pvc_obj.delete()
+        self.sa_name.delete()
+
+    @pytest.mark.parametrize("copies", [5])
+    def test_selinux_relabel(self, copies):
+        """
+        Steps:
+            1. Create multiple cephfs pvcs and 100K files each across multiple nested directories
+            2. Run the IOs for some random files and take md5sum for them
+            3. Apply the fix/solution as mentioned in the “Existing PVs” section
+            4. Restart the pods which are hosting cephfs files in large numbers
+            5. Check data integrity.
+            6. Check for relabeling - this should not be happening.
+
+        """
+        self.project_namespace = constants.OPENSHIFT_STORAGE_NAMESPACE
+        # Create pvc
+        self.pvc_obj = helpers.create_pvc(
+            namespace=self.project_namespace,
+            sc_name=constants.DEFAULT_STORAGECLASS_CEPHFS,
+            size="20Gi",
+        )
+
+        # Create deployment pod
+        self.pod_obj = self.create_deploymentconfig_pod(
+            command=["/opt/multiple_files.sh"],
+            command_args=[f"{copies}", "/mnt"],
+        )
+        log.info(f"files copied to pod {self.pod_obj}")
+        self.pod_selector = self.pod_obj.labels.get("deploymentconfig")
+
+        # Check data integrity before applying selinux relabeling solution
+        start_time1 = time.time()
+        self.data_integrity_check(self.pod_obj, self.project_namespace)
+        end_time1 = time.time()
+        total_time1 = end_time1 - start_time1
+        log.info(f"Time taken by pod to restart is  {total_time1}")
+
+        # Apply the fix/solution for “Existing PVCs”
```

Comment:
This code has to be used for other selinux tests as well, please create a function so that it can be reused